### PR TITLE
prevent fork deleted component

### PR DESCRIPTION
**Pull request by chennan47**
<b>Purpose</b>
Prevent project's deleted component to be forked.

<b>Changes</b>
Before change:
screen shot 2015-06-07 at 6 10 08 pm
After change:
The deleted component will no longer be forked.

### CODE REVIEW

File: website/project/model.py [hunk 1]
```diff
@@ -1583,6 +1583,9 @@ def fork_node(self, auth, title='Fork of '):
 
         original = self.load(self._primary_key)
 
+        if original.is_deleted:
+            raise NodeStateError('Cannot fork deleted node.')
```
Comment (sloria) [inline website/project/model.py]: Does this error need to be handled in the views?

**Comment by sloria** [comment]
Please add a regression test for this change.

**Comment by sloria** [comment]
Pass finished. :point_up:

**Comment by chennan47** [comment]
@sloria regression tests added, rdy for another pass.

**Pull request merged**